### PR TITLE
Extending unit test by validation on error message

### PR DESCRIPTION
**Pull request by ASkrypka**
- add unit test:NullOrEmpty
- add unit test:NullOrWhiteSpace
- add unit test:OutOfRangeForDecimal
- add unit test:OutOfRangeForDouble
- add unit test:OutOfRangeForEnum
- add unit test:OutOfRangeForEnumerableDecimal
- add unit test:OutOfRangeForEnumerableDouble
- add unit test:OutOfRangeForEnumerableFloat
- add unit test:OutOfRangeForEnumerableInt
- add unit test:OutOfRangeForEnumerableLong
- add unit test:OutOfRangeForEnumerableShort
- add unit test:OutOfRangeForEnumerableTimeSpan
- add unit test:OutOfRangeForFloat
- add unit test:OutOfRangeForInt
- add unit test:OutOfRangeForInvalidInput
- add unit test:OutOfRangeForShort
- add unit test:OutOfRangeForTimeSpan
- add unit test:OutOfSQLDateRange
- add unit test:Zero
- add unit test:Default
- add unit test:InvalidFormat
- add unit test:Negative
- add unit test:NegativeOrZero
- add unit test:Null
- fix spacing

### CODE REVIEW

**Review by apm1grb — COMMENTED** [review]
cannot follow: commit message says "fix spacing", but change shows only adjusting version number

File: test/GuardClauses.UnitTests/GuardAgainstOutOfRangeForEnumerableFloat.cs [hunk 1]
```diff
@@ -22,13 +22,29 @@ public void ThrowsGivenOutOfRangeValue(IEnumerable<float> input, float rangeFrom
...
         public void ThrowsGivenInvalidArgumentValue(IEnumerable<float> input, float rangeFrom, float rangeTo)
         {
             Assert.Throws<ArgumentException>(() => Guard.Against.OutOfRange(input, nameof(input), rangeFrom, rangeTo));
         }
 
+        [Theory]
+        [ClassData(typeof(IncorrectRangeClassData))]
+        public void ThrowsSelfOwnErrorMessageGivenInvalidArgumentValue(IEnumerable<float> input, float rangeFrom, float rangeTo)
+        {
+           var exception = Assert.Throws<ArgumentException>(() => Guard.Against.OutOfRange(input, nameof(input), rangeFrom, rangeTo, "selfOwnErrorMessage"));
```
Comment (apm1grb) [inline test/GuardClauses.UnitTests/GuardAgainstOutOfRangeForEnumerableFloat.cs:44]: something is wrong with indentation

File: test/GuardClauses.UnitTests/GuardAgainstOutOfRangeForEnumerableInt.cs [hunk 2]
```diff
@@ -22,13 +22,29 @@ public void ThrowsGivenOutOfRangeValue(IEnumerable<int> input, int rangeFrom, in
             Assert.Throws<ArgumentOutOfRangeException>(() => Guard.Against.OutOfRange(input, nameof(input), rangeFrom, rangeTo));
         }
 
+        [Theory]
+        [ClassData(typeof(IncorrectClassData))]
+        public void ThrowsSelfOwnErrorMessageGivenOutOfRangeValue(IEnumerable<int> input, int rangeFrom, int rangeTo)
+        {
+           var exception = Assert.Throws<ArgumentOutOfRangeException>(() => Guard.Against.OutOfRange(input, nameof(input), rangeFrom, rangeTo, "selfOwnErrorMessage"));
```
Comment (apm1grb) [inline test/GuardClauses.UnitTests/GuardAgainstOutOfRangeForEnumerableInt.cs:29]: again, indentation --> please use ReSharper and option "Format Document" to avoid such flaws

File: test/GuardClauses.UnitTests/GuardAgainstOutOfRangeForEnumerableTimeSpan.cs [hunk 3]
```diff
@@ -31,6 +31,18 @@ public void ThrowsGivenOutOfRangeValue(IEnumerable<int> input, int rangeFrom, in
...
 
+        [Theory]
+        [ClassData(typeof(IncorrectClassData))]
+        public void ThrowsSelfOwnErrorMessageGivenOutOfRangeValue(IEnumerable<int> input, int rangeFrom, int rangeTo)
+        {
+            var inputTimeSpan = input.Select(i => TimeSpan.FromSeconds(i));
+            var rangeFromTimeSpan = TimeSpan.FromSeconds(rangeFrom);
+            var rangeToTimeSpan = TimeSpan.FromSeconds(rangeTo);
+
+           var exception = Assert.Throws<ArgumentOutOfRangeException>(() => Guard.Against.OutOfRange(inputTimeSpan, nameof(inputTimeSpan), rangeFromTimeSpan, rangeToTimeSpan, "selfOwnErrorMessage"));
```
Comment (apm1grb) [inline test/GuardClauses.UnitTests/GuardAgainstOutOfRangeForEnumerableTimeSpan.cs:42]: again indentation

File: test/GuardClauses.UnitTests/GuardAgainstNegative.cs [hunk 4]
```diff
@@ -27,43 +27,86 @@ public void DoesNothingGivenPositiveValue()
...
         [Fact]
         public void ThrowsGivenNegativeTimeSpanValue()
         {
             Assert.Throws<ArgumentException>(() => Guard.Against.Negative(TimeSpan.FromSeconds(-1), "negative"));
         }
 
+        [Fact]
+        public void ThrowsSelfOwnErrorMessageGivenNegativeTimeSpanValue()
+        {
+
```
Comment (apm1grb) [inline test/GuardClauses.UnitTests/GuardAgainstNegative.cs:105]: unwanted empty line

File: test/GuardClauses.UnitTests/GuardAgainstOutOfRangeForDouble.cs [hunk 5]
```diff
@@ -31,8 +31,8 @@ public void ThrowsGivenOutOfRangeValue(double input, double rangeFrom, double ra
         [InlineData(4.0, 1.0, 3.0)]
         public void ThrowsSelfOwnErrorMessageGivenOutOfRangeValue(double input, double rangeFrom, double rangeTo)
         {
-            var ex = Assert.Throws<ArgumentOutOfRangeException>(() => Guard.Against.OutOfRange(input, "index", rangeFrom, rangeTo, "selfOwnErrorMessage"));
-            Assert.Contains("selfOwnErrorMessage", ex.Message);
+            var exception = Assert.Throws<ArgumentOutOfRangeException>(() => Guard.Against.OutOfRange(input, "index", rangeFrom, rangeTo, "selfOwnErrorMessage"));
+            Assert.Contains("selfOwnErrorMessage", exception.Message);
```
Comment (apm1grb) [inline test/GuardClauses.UnitTests/GuardAgainstOutOfRangeForDouble.cs]: Content of the commit message doesn't reflect what the change is about

**Review by apm1grb — COMMENTED** [review]
Commit message doesn't reflect commit content

- there is one added unit test
- but also several unit test adjustments

--> in the future: split the unit-test into 2 (or more) commits with focussed content & message

(targets the second last commit of this PR)
